ports/testport.js

```diff
@@ -185,7 +185,27 @@ TestPort.prototype.write = function (buf) {
         this._holding_registers[address] = value;
     }
 
+    // function code 15
+    if (functionCode == 15) {
+        var address = buf.readUInt16BE(2);
+        var length = buf.readUInt16BE(4);
+
+        // if length is bad, ignore message
+        if (buf.length != 7 + Math.ceil(length / 8) + 2) {
+            return;
+        }
+
+        // build answer
+        buffer = new Buffer(8);
+        buffer.writeUInt16BE(address, 2);
+        buffer.writeUInt16BE(length, 4);
 
+        // write coils
+        var byteCounts = Math.ceil(length / 8);
+        for (var i = 0; i < byteCounts; i++) {
+            this._coils &= buf.readUInt8(7 + i);
```

Comment:
Yes, the coil address would miss if another function code 15 comes. I care little at here. I would fix it, thanks for your correction :-)